### PR TITLE
MM-19234 Migrate tests from model/team_member_test.go to use testify

### LINKED ISSUE #12636: Migrate tests from "model/team_member_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `model/team_member_test.go` file.

The expected way to migrate this is to go to the `model/team_member_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19234

### PR DESCRIPTION
**Pull request by Pensu**
#### Summary

Replace t.Fatal with require from testify.

#### Ticket Link

Fixes #12636 
JIRA: https://mattermost.atlassian.net/browse/MM-19234

### CODE REVIEW

File: model/team_member_test.go [hunk 1]
```diff
@@ -6,29 +6,27 @@ package model
...
 }
 
 func TestTeamMemberIsValid(t *testing.T) {
 	o := TeamMember{}
 
-	if err := o.IsValid(); err == nil {
-		t.Fatal("should be invalid")
-	}
+	err := o.IsValid()
+	require.Nil(t, err, "should be invalid")
```
Comment (gabrieljackson) [inline model/team_member_test.go]: I believe this require statement is the opposite of what we want. Before, we would `fatal` if `err == nil` and here we are requiring nil.

Nit: also, we could just make this one line like so:
```suggestion
	require.Error(t, o.IsValid(), "should be invalid")	
```

File: model/team_member_test.go [hunk 2]
```diff
@@ -6,29 +6,27 @@ package model
...
-	}
+	err := o.IsValid()
+	require.Nil(t, err, "should be invalid")	
 
 	o.TeamId = NewId()
-	if err := o.IsValid(); err == nil {
-		t.Fatal("should be invalid")
-	}
+	err = o.IsValid()
+	require.Nil(t, err, "should be invalid")
```
Comment (gabrieljackson) [inline model/team_member_test.go]: Same as above.

File: model/team_member_test.go (hunk 1 above)
Comment (hmhealey) [reply model/team_member_test.go]: Good catch. That explains why the tests were failing

**Comment by hmhealey** [comment]
@Pensu It looks like you'll need to re-sign the CLA with your name capitalized to match GitHub. It seems that the form is case sensitive since I can see "pensu" on the list of approved contributors, but it is looking for "Pensu"

**Comment by Pensu** [comment]
Hi @gabrieljackson, I have updated the PR. Not sure about the failure, it says PR needs to sync and merge.

Hi @hmhealey, I have resubmitted the CLA form with the updated name. Please check.

**Comment by hmhealey** [comment]
The build is failing due to some unrelated issues on the web app tests. I'll update the branch to restart those.

**Pull request merged**